internal/service/config/nginx_config_service.go

```diff
@@ -147,12 +149,21 @@ func (n *Nginx) Validate() error {
 
 func (n *Nginx) Apply() error {
 	slog.Debug("Applying NGINX config")
-	exePath := n.instance.GetMeta().GetNginxMeta().GetExePath()
-	out, err := n.executor.RunCmd(exePath, "-s", "reload")
+
+	processID := n.instance.GetMeta().GetNginxMeta().GetProcessId()
+
+	intProcessID, err := strconv.Atoi(processID)
+	if err != nil {
+		slog.Error("Invalid NGINX master process ID", "process_id", processID, "error", err)
+		return err
+	}
+
+	err = n.executor.KillProcess(intProcessID, syscall.SIGHUP)
```

Comment:
will we ever do a kill process with a different syscall? I think it's fine to encapsulate it in the KillProcess function for now and we can extend it later

Reply:
yeah probably not. Let me make that update there now